.github/workflows/pipeline-deploy-app-non-prod.yaml

```diff
@@ -0,0 +1,230 @@
+name: Application Deployment Pipeline - Non Prod Environments
+
+on:
+  push:
+    branches:
+      - main
+      - 'task/*'
+  pull_request:
+    types:
+      - labeled
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: "Specify the tag to be used for deployment"
+        required: true
+        type: string
+      environment:
+        description: 'Deployment environment'
+        required: true
+        default: 'dev'
+        type: choice
+        options:
+          - dev
+          - test
+          - int
+
+jobs:
+
+  metadata:
+    name: "Get Metadata"
+    uses: NHSDigital/uec-dos-management/.github/workflows/metadata.yaml@latest
+
+  generate-tag-name:
+    if: ${{ (needs.metadata.outputs.environment == 'test' || needs.metadata.outputs.environment == 'int') && github.event.action == 'labeled' }}
+    name: "Generate Tag Name"
+    needs:
+      [
+        metadata,
+      ]
+    uses: NHSDigital/uec-dos-management/.github/workflows/generate-tag-name.yaml@latest
+    with:
+      workspace: ${{ needs.metadata.outputs.workspace }}
+      tag_type: ${{ needs.metadata.outputs.environment }}
+      use_commit_hash: "yes"
+      commit_hash: ${{ needs.metadata.outputs.artefact_commit_hash }}
+
+  tag-repository:
+    if: ${{ (needs.metadata.outputs.environment == 'test' || needs.metadata.outputs.environment == 'int') && github.event.action == 'labeled' }}
+    name: "Tag Repository"
+    needs:
+      [
+        metadata,
+        generate-tag-name,
+      ]
+    uses: NHSDigital/uec-dos-management/.github/workflows/tag-repository.yaml@latest
+    with:
+      tag_to_push: ${{ needs.generate-tag-name.outputs.tag_name }}
+      tag_overwrite: "yes"
+
+  quality-check:
+    if: ${{ needs.metadata.outputs.environment == 'dev' }}
+    name: "Quality Check"
+    uses: NHSDigital/uec-dos-management/.github/workflows/quality-checks.yaml@latest
+    needs:
+      [
+        metadata,
+      ]
+    with:
+      environment: ${{ needs.metadata.outputs.environment }}
+      workspace: ${{ needs.metadata.outputs.workspace }}
+      stacks: '["github-runner", "terraform_management", "application", "data_migration", "domain-test"]'
+    secrets: inherit
+
+  unit-tests:
+    if: ${{ needs.metadata.outputs.environment == 'dev' }}
+    name: Unit Test
+    needs:
+      [
+        metadata,
+        quality-check,
+      ]
+    uses: NHSDigital/uec-dos-management/.github/workflows/unit-test.yaml@latest
+    with:
+      env: ${{ needs.metadata.outputs.environment }}
+    secrets: inherit
+
+  build-application-artefacts:
+    if: ${{ needs.metadata.outputs.environment == 'dev' }}
+    name: Build application artefacts and push to storage
+    needs:
+      [
+        metadata,
```

Comment:
Building of artefacts should depend on unit-tests.

Reply:
Agreed - will make dev team aware of restriction